### PR TITLE
Prevent labels from being overwritten during mutations

### LINKED ISSUE #1039: Provisioner overrides labels when mutating clusters
If you spin up a cluster with the provisioner, it populates your plan file. If you add labels to nodes using the plan and apply the labels, then mutate the cluster (either by adding or removing nodes), your labels will be overwritten with nothing when provisioning completes.

### PR DESCRIPTION
**Pull request by emmetthitz**
Fixes #1039.

### CODE REVIEW

**Comment by dkoshkin** [comment]
Need to rebase this with just the commits for this PR

File: pkg/install/plan_types.go [hunk 1]
```diff
@@ -505,9 +505,60 @@ type Node struct {
...
+				ret = append(ret, Node{
+					IP:         l[i].IP,
+					InternalIP: l[i].InternalIP,
+					Host:       l[i].Host,
+					Labels:     mapMerge(l[i].Labels, r[j].Labels),
+					KubeletOptions: KubeletOptions{
+						Overrides: mapMerge(l[i].KubeletOptions.Overrides, r[j].KubeletOptions.Overrides),
+					},
+				})
+				//significantly faster than just axing the single element
```
Comment (dkoshkin) [inline pkg/install/plan_types.go]: I think preserving the order would be better in this case, especially since these slices will be pretty small.

File: pkg/install/plan_types.go [hunk 2]
```diff
@@ -505,9 +505,60 @@ type Node struct {
...
+	for k, v := range r {
+		if _, ok := ret[k]; ok {
+			continue
+		}
+		ret[k] = v
+	}
+	return ret
+}
+
+func merge(l, r []Node) []Node {
```
Comment (dkoshkin) [inline pkg/install/plan_types.go]: Can you add some unit tests, it would be good to test some edge cases and make sure everything works as expected.

Comment (emmetthitz) [reply pkg/install/plan_types.go]: Sure thing.

File: pkg/install/plan_types.go (hunk 1 above)
Comment (emmetthitz) [reply pkg/install/plan_types.go]: slice r ultimately ends up empty no matter what. preserving its order shouldn't matter. are you thinking of some case in particular where this might be a problem?

Comment (dkoshkin) [reply pkg/install/plan_types.go]: Maybe Im misunderstanding how this is going to work, but I expect when modifying the node count and TF adds a new node the new order of the nodes will remain the same with an additional one at the end.

Comment (emmetthitz) [reply pkg/install/plan_types.go]: Yeah, that shouldn't be affected.

Comment (dkoshkin) [reply pkg/install/plan_types.go]: Great, would be awesome to have a unit test showing that.

**Comment by dkoshkin** [comment]
LGTM!
